WireDomain/Sources/WireDomain/Notifications/VerifyUserSession.swift

```diff
@@ -0,0 +1,126 @@
+//
+// Wire
+// Copyright (C) 2025 Wire Swiss GmbH
+//
+// This program is free software: you can redistribute it and/or modify
+// it under the terms of the GNU General Public License as published by
+// the Free Software Foundation, either version 3 of the License, or
+// (at your option) any later version.
+//
+// This program is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+// GNU General Public License for more details.
+//
+// You should have received a copy of the GNU General Public License
+// along with this program. If not, see http://www.gnu.org/licenses/.
+//
+
+import WireAPI
+import WireDataModel
+
+/// Observes pending events, process them and generates new notifications content.
+struct VerifyUserSession {
+
+    enum Constants {
+        static let cookieName = "zuid"
+    }
+
+    // MARK: - Error
+
+    enum Failure: Error {
+        case coreDataMissingSharedContainer
+        case coreDataMigrationRequired
+        case userUnauthenticated
+        case missingUserClient
+        case unableToLoadStores(Error)
+    }
+
+    // MARK: - Properties
+
+    typealias NotificationHandler = (UNMutableNotificationContent) -> Void
+
+    private let pullEventsServiceProvider: any PullEventsServiceProvider
+    private let cookieStorage: any CookieStorageProtocol
+    private let coreData: any CoreDataStackProtocol
+
+    init(
+        pullEventsServiceProvider: any PullEventsServiceProvider,
+        cookieStorage: any CookieStorageProtocol,
+        coreData: any CoreDataStackProtocol
+    ) {
+        self.pullEventsServiceProvider = pullEventsServiceProvider
+        self.cookieStorage = cookieStorage
+        self.coreData = coreData
+    }
+
+    /// Ensures user is properly authenticated.
+    /// - parameters
+    ///     - userID: The user ID to verify the authentication for.
+    ///     - handler: Completion block called if the user is authenticated.
+
+    func verify(
+        userID: UUID,
+        then completion: () async throws -> Void
+    ) async throws {
+        let cookies = try await cookieStorage.fetchCookies()
+        var hasExpirationDate = false
+
+        for cookie in cookies where cookie.name == Constants.cookieName {
+            hasExpirationDate = cookie.expiresDate != nil
```

Comment:
question: are we looking for at least one cookie with an expiration date? If so, we might set this to `true` and then `false` again. Also... shouldn't we check if the expiration date has not been reached?